argyrogounari/assignment3/Main.java

```diff
@@ -0,0 +1,65 @@
+/**
+* Given a grid of letters and a dictionary, find all the words from the dictionary that can be formed in the grid.
+* The rules for forming a word:
+* You can start at any position.
+* You can move to one of the 8 adjacent cells (horizontally/vertically/diagonally).
+* You can't visit the same cell twice in the same word.
+*
+* @author  Argyro Gounari
+* @version 1.0
+* @since   12-06-2018 
+*/
+public class Main {
+
+	public static void main(String[] args) {
+		
+		Dictionary theDictionary = new Dictionary();
+		
+		char grid[][] = new char[2][3];
+		grid[0][0] = 'A';
+		grid[0][1] = 'A';
+		grid[0][2] = 'R';
+		grid[1][0] = 'T';
+		grid[1][1] = 'C';
+		grid[1][2] = 'D';
+		// A A R
+		// T C D
+		
+		wordsFormed(theDictionary, grid);
+		
+	}
+	
+	// Print words in grid
+	public static void wordsFormed(Dictionary theDictionary, char[][] grid) {
+		
+		String str = "";
+		Boolean seen[][] = new Boolean[2][3];
```

Comment:
Make sure you use the size of the grid as the size of the seen array, otherwise a different sized grid would fail:
boolean[][] seen = new boolean[grid[0].length][grid.length];